packages/design-system/src/components/Pagination/Pagination.tsx

```diff
@@ -285,19 +262,19 @@ function Pagination({
         {startLabelText ?? t('pagination.startLabelText')}
       </Button>
 
-      {isMobile || compact ? (
-        <span
-          className="ds-c-pagination__page-count"
-          dangerouslySetInnerHTML={{
-            __html: t('pagination.pageXOfY', {
-              number: `<strong>${currentPage}</strong>`,
-              total: `<strong>${totalPages}</strong>`,
-            }),
-          }}
-        />
-      ) : (
-        <ul role="list">{pages}</ul>
-      )}
+      <span
+        className={`ds-c-pagination__page-count${!compact ? ' ds-u-display--none' : ''}`}
```

Comment:
Right now there's a case where both are shown at once, whereas previously the two views were in a mututally exclusive ternary.

<img width="778" alt="Screenshot 2023-03-02 at 11 32 15 AM" src="https://user-images.githubusercontent.com/7595652/222533543-b89e7c54-7f70-464b-bbd9-cd3b8ebe2a1e.png">

I think the logic could be further pushed to CSS by removing the ternaries on these individual elements and instead placing a `ds-c-pagination--compact` conditionally on the top level component. That would allow the CSS to take advantage of that class to say "if that class exists, hide the list and show the page count; if it doesn't exist, use the media queries to govern which one is shown". If we don't want to have to use `:not()` selectors, we could have an opposite class to `--compact` like `--wide` or something. Moving it all to CSS also grants applications the opportunity to override the behavior if needed.

Reply:
Good points, updated, thanks.